components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -962,19 +962,33 @@ func (p *pg) CreateRule(ctx context.Context, rule *v2.Rule) (*v2.Rule, error) {
 	if err != nil {
 		return nil, p.processError(err)
 	}
+	
+	row:= tx.QueryRowContext(ctx,
+		`SELECT query_rule_table_associations($1);`, rule.ID)
+	if err != nil {
+		return nil, p.processError(err)
+	}
+	var associations []string
+	if err := row.Scan(pq.Array(&associations)); err != nil {
+		return nil, p.processError(err)
+	}
+	// If any associations return, then the rule already exists in current, staged, or both tables
+	if len(associations) > 0 {
```

Comment:
Some bits strike me as a bit odd:
1. we explicitly convert whatever is returned into an _array_ in the SQL call, so that only one row is returned -- the array. This is a bit odd, I'd expect this to just return the `TableName`s as multiple rows, if there's multiple of them.
2. if we only care about _any of them existing_ (`len(associations) > 0`), we might as well build the query for just that, instead of shuffling around a `[]string` we're not interested in. 

(2.) with the change of (1.) could become something using `EXISTS`, I suppose. 🤔 Alternatively, we could perhaps use `ExecContext` (instead of `QueryRowContext`) and check `(Result).RowsAffected`? (Not sure if that second proposal works, though!)

Reply:
For context, at this juncture, we only care about whether it already exists (for `Create`). We will, however, have to look at the returned values for calls such as `Update` (if it exists in the `applied` table only, we'll create a new row in `staged` . . . if it exists in `staged`, we'll update that row.) So we'll eventually be interested in the `[]string`. We don't need it here, really, though.

Totally open to better ideas on how to do ⬆️ though. (An array seemed like a better data structure for this purpose but perhaps we should just be examining the rows returned?)

Reply:
Ah, I see. Thanks for the context. It's fine I think. 👍